### PR TITLE
Member leave group

### LINKED ISSUE #95: Update text on members page for member removal
### Problem

On the members page for both groups and projects, currently the text to remove a member from the group or project just has the text `Remove`. We want to update this to `Remove member`, if the member is not the currently logged in user, and to either read `Leave group` or `Leave project`, if the member is the currently logged in user. The confirmation message would also need to be updated to be in-line with this new text for the case of current user is the member. 

### Implementation

- Take a look at the relevant views `app/views/groups/members/index.html.erb` and `app/views/projects/members/index.html.erb`
- Within the views, look at the viral_dropdown section which has the links for actions (dropdown_item).
- Make a note of the i18n translation keys for `label` and the `turbo_confirm`
- In the translation file `en.yml`, update the value to be in-line with the new text `Remove member`
- Also, add in new translation key/values for `Leave group`, `Leave project`, and confirmation message to match these new values
- Once the `en.yml` file has been updated you will need to conditionally display the correct translation key if the current user is the member or not
- You will also need to update the system tests to ensure the correct text is displayed

### Additional Details

Currently this is what the members pages look like. Members pages are accessible from within a group and within a project

Group members page: (Change to `Leave Group` as the currently logged in user is `admin@email.com` and the user being removed is this user)

image

Project members page: (Change to `Remove member` as the currently logged in user is `admin@email.com` and the user being removed is not this user)

image

- To view the current confirmation message you can click on the `Remove` for a member

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds functionality for a member to be able to leave a group or project on their own without needing an owner or maintainer to remove them.

Due to a bad rebase and force update of commits PR #114 ended up with 0 commits and was closed. Opened this PR to replace it.
Resolves #95 and #96 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Logged in as user that is removing themselves from the group:**

image

image

**Removing a member that is not the current logged in user**

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@example.com`
2. Go the the `Bacillus` group and view it's members
3. Click the ellipsis for the `admin@example.com` user and you should see a link `Leave group`
4. In the confirmation you should see `Are you sure you want to remove yourself from the group?`
5. Try step 3 for another user in the listing and you should see a link `Remove member`
6. In the confirmation you should see `Are you sure you want to remove this member from the group?`
7. Try out the same steps but for a project. Projects from the seed data only have user6 through user8 as members so you will need to log in as one of those users.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the PR acceptance checklist for this PR.

### CODE REVIEW

File: app/views/groups/members/_current_user_is_member.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<%= viral_dropdown(icon: "ellipsis_vertical",
+                                     classes: "member-settings-ellipsis",
```
Comment (joshsadam) [inline app/views/groups/members/_current_user_is_member.html.erb:2]: **question:** Why do you need this class?

**Review by joshsadam — COMMENTED** [review]
This looks really good, just one question.

File: app/views/groups/members/_current_user_is_member.html.erb (hunk 1 above)
Comment (deepsidhu85) [reply app/views/groups/members/_current_user_is_member.html.erb]: This was added pretty much at the beginning of when the members pages were developed. It was used to get the ellipsis in the tests. However we no longer use the user defined classes/ids to find stuff on the page. I've removed them in 09df5454052849947625f43e56f308b9b188e427

File: app/views/groups/members/index.html.erb [hunk 2]
```diff
@@ -82,27 +82,12 @@
...
-                        },
-                        aria: {
-                          label:
-                            t(
-                              :"groups.members.index.actions.link_remove_aria_label",
-                              member: member.user.email
-                            )
-                        }
-                      ) %>
+                    <% if current_user == member.user %>
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply app/views/groups/members/index.html.erb:85]: Thanks!

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**